BridgeSDK/BridgeAPI/Categories/NSError+SBBAdditions.m

```diff
@@ -75,54 +84,85 @@ + (NSError*) generateSBBErrorForStatusCode:(NSInteger)statusCode data: (id) data
 {
-  return [NSError errorWithDomain:SBB_ERROR_DOMAIN code:kSBBServerNotAuthenticated userInfo:@{NSLocalizedDescriptionKey: @"Server says: not authenticated. Please authenticate."}];
+  return [NSError errorWithDomain:SBB_ERROR_DOMAIN code:SBBErrorCodeServerNotAuthenticated
+                         userInfo:@{NSLocalizedDescriptionKey: NSLocalizedString(@"Server says: not authenticated. Please authenticate.",
+                                                                                 @"Error Description: not authenticated")}];
 }
 
 + (NSError *)generateSBBNotAFileURLErrorForURL:(NSURL *)url
 {
-  NSString *desc = [NSString stringWithFormat:@"Not a valid file URL:\n%@", url];
-  return [NSError errorWithDomain:SBB_ERROR_DOMAIN code:kSBBNotAFileURL userInfo:@{NSLocalizedDescriptionKey: desc}];
+  NSString *localizedFormat = NSLocalizedString(@"Not a valid file URL:\n%@", @"Error Description: not a valid url");
+  NSString *desc = [NSString stringWithFormat:localizedFormat, url];
+  return [NSError errorWithDomain:SBB_ERROR_DOMAIN code:SBBErrorCodeNotAFileURL
+                         userInfo:@{NSLocalizedDescriptionKey: desc}];
 }
 
 + (NSError *)generateSBBTempFileErrorForURL:(NSURL *)url
 {
-  NSString *desc = [NSString stringWithFormat:@"Error copying file at URL to temp file:\n%@", url];
-  return [NSError errorWithDomain:SBB_ERROR_DOMAIN code:kSBBTempFileError userInfo:@{NSLocalizedDescriptionKey: desc}];
+  NSString *localizedFormat = NSLocalizedString(@"Error copying file at URL to temp file:\n%@", @"Error Description: error copying file");
+  NSString *desc = [NSString stringWithFormat:localizedFormat, url];
+  return [NSError errorWithDomain:SBB_ERROR_DOMAIN code:SBBErrorCodeTempFileError userInfo:@{NSLocalizedDescriptionKey: desc}];
 }
 
 + (NSError *)generateSBBTempFileReadErrorForURL:(NSURL *)url
 {
-  NSString *desc = [NSString stringWithFormat:@"Error reading temp file for original file URL:\n%@", url];
-  return [NSError errorWithDomain:SBB_ERROR_DOMAIN code:kSBBTempFileError userInfo:@{NSLocalizedDescriptionKey: desc}];
+  NSString *localizedFormat = NSLocalizedString(@"Error reading temp file for original file URL:\n%@", @"Error Description: error reading temp file");
+  NSString *desc = [NSString stringWithFormat:localizedFormat, url];
+  return [NSError errorWithDomain:SBB_ERROR_DOMAIN code:SBBErrorCodeTempFileError userInfo:@{NSLocalizedDescriptionKey: desc}];
 }
 
 + (NSError *)generateSBBObjectNotExpectedClassErrorForObject:(id)object expectedClass:(Class)expectedClass
 {
-  NSString *desc = [NSString stringWithFormat:@"Object '%@' is of class %@, expected class %@", object, NSStringFromClass([object class]), NSStringFromClass(expectedClass)];
-  return [NSError errorWithDomain:SBB_ERROR_DOMAIN code:kSBBObjectNotExpectedClass userInfo:@{NSLocalizedDescriptionKey: desc}];
+  NSString *localizedFormat = NSLocalizedString(@"Object '%1$@' is of class %2$@, expected class %3$@", @"Error Description");
```

Comment:
Well, this particular error is one that shouldn't happen in production and should never be shown to an end user (maybe used in error logging though). So maybe something like "Message for internal error logging when an object of unexpected class is received from the server, to be filled in with..." etc. as in my previous comment.